src/LumiSpecHomoCAL_geo.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ Written by - Aranya Giri
+ University of Houston
+ Start Date - 10/31/2022
+ Homogeneous PbWO4 (EM Calorimeter) Pair Spectrometer
+ Note : Solid, Logical & Physical Volumes are the three component of detector,
+ The following abbriviations will reflect the proper usage of these 3.
+*/
+
+// Essential Header Files
+#include "DD4hep/DetFactoryHelper.h"
+//#include "GeometryHelpers.h"
+#include <XML/Helper.h>
+#include <algorithm>
+#include <iostream>
+#include <math.h>
+#include <tuple>
+
+//Additional Header Files
+
+using namespace std;
+using namespace dd4hep;
+//using Point = ROOT::Math::XYPoint;
+
+//Definition of function to build the modules
+tuple<Volume, Position> build_specHomoCAL_module(const Detector& lccdd, const xml::Component& mod_x, SensitiveDetector& sens);
+
+//Driver Function
+static Ref_t create_detector(Detector& lccdd, xml_h e, SensitiveDetector sens){
+
+  sens.setType("calorimeter");
+
+  xml_det_t 	x_det 		= 	e;
+  xml_dim_t 	x_dim 		= 	x_det.dimensions();
+  xml_dim_t	x_pos		= 	x_det.position();
+  xml_dim_t	x_rot		= 	x_det.rotation();
+
+  string		detName		= 	x_det.nameStr();
+  int		detID		=	x_det.id();
+
+  double         	env_dx          = 	x_dim.x();
+  double         	env_dy          = 	x_dim.y();
+  double         	env_dz          = 	x_dim.z();
+
+  //Create Envolope
+  Box envSolid(env_dx/2.0, env_dy/2.0, env_dz/2.0);
+
+  Volume envLogic(detName+ "_envLogic", envSolid, lccdd.material("Vacuum") );
```

Comment:
Is this actually contained in a vacuum? I would look at defining this as an Assembly of the detector sub-components.

Reply:
Changed to an assembly.